systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java

```diff
@@ -292,4 +296,29 @@ void testCruiseControlIntraBrokerBalancing(ExtensionContext extensionContext) {
         KafkaRebalanceStatus kafkaRebalanceStatus = KafkaRebalanceResource.kafkaRebalanceClient().inNamespace(testStorage.getNamespaceName()).withName(testStorage.getClusterName()).get().getStatus();
         assertThat(kafkaRebalanceStatus.getOptimizationResult().get("provisionStatus").toString(), containsString("UNDECIDED"));
     }
+
+    @ParallelNamespaceTest
+    void testCruiseControlIntraBrokerBalancingWithoutSpecifyingJBODStorage(ExtensionContext extensionContext) {
+        final TestStorage testStorage = new TestStorage(extensionContext);
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaWithCruiseControl(testStorage.getClusterName(), 3, 3).build());
+        resourceManager.createResource(extensionContext, KafkaRebalanceTemplates.kafkaRebalance(testStorage.getClusterName())
+            .editOrNewSpec()
+                .withRebalanceDisk(true)
+            .endSpec()
+            .build());
+
+        final String coPodName = KubeClusterResource.kubeClient(INFRA_NAMESPACE).getClusterOperatorPodName();
+        final String coLog = StUtils.getLogFromPodByTime(INFRA_NAMESPACE, coPodName, STRIMZI_DEPLOYMENT_NAME, "60s");
+
+        // CO logs must contain warning message
+        assertThat(coLog, CoreMatchers.containsString("Intra-broker balancing only applies to Kafka deployments that use JBOD storage with multiple disks."));
+
+        KafkaRebalanceUtils.waitForKafkaRebalanceCustomResourceState(testStorage.getNamespaceName(), testStorage.getClusterName(), KafkaRebalanceState.ProposalReady);
+        // Strimzi will force an inter-broker balance when rebalanceDisk=`true` and a JBOD configuration is not found.
+        // The provision status should be "RIGHT_SIZED" because Strimzi is preforming an inter-broker balance here and the
+        // cluster should not be under or over provisioned by this rebalance.
+        KafkaRebalanceStatus kafkaRebalanceStatus = KafkaRebalanceResource.kafkaRebalanceClient().inNamespace(testStorage.getNamespaceName()).withName(testStorage.getClusterName()).get().getStatus();
+        assertThat(kafkaRebalanceStatus.getOptimizationResult().get("provisionStatus").toString(), containsString("RIGHT_SIZED"));
+    }
```

Comment:
@kyguy @see-quick I don't get this ... you request intra-broker rebalance but do not have JBOD storage. So why does it not fail and just write some warning? I expected that all the elaborate detection in the code was to fail the rebalance in this case. Not just decide to do a normal rebalance.

Reply:
You are right, this does not make sense, this was an oversight on my part when writing the feature, let me put in a fix for it and we can adjust this to test this behavior

Reply:
So, should we merge this and you fix the test together with the code? Or should we keep it opened and @see-quick updates it later after you fix the code?

Reply:
We can merge it and I'll fix the test together with the code, I'll open an issue for it today and link it here

Reply:
Issue here #6383